raphtory/src/python/graph/edge.rs

```diff
@@ -599,6 +677,174 @@ impl PyEdges {
         (move || edges().id()).into()
     }
 
+    /// Returns all timestamps of edges, when an edge is added or change to an edge is made.
+    ///
+    /// Returns:
+    ///    A list of timestamps.
+    ///
+
+    fn history(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().history()).into()
+    }
+
+    /// Get the start time of all edges
+    ///
+    /// Returns:
+    /// The start time of all edges
+    #[getter]
+    fn start(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().start()).into()
+    }
+
+    /// Get the start date time of all edges
+    ///
+    /// Returns:
+    /// The start date time of all edges
+    #[getter]
+    fn start_date_time(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().start_date_time()).into()
+    }
+
+    /// Get the end time of all edges
+    ///
+    /// Returns:
+    /// The end time of all edges
+    #[getter]
+    fn end(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().end()).into()
+    }
+
+    /// Get the end date time of all edges
+    ///
+    /// Returns:
+    ///  The end date time of all edges
+    #[getter]
+    fn end_date_time(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().end_date_time()).into()
+    }
+
+    /// Get the layer name that all edges belong to - assuming they only belong to one layer
+    ///
+    /// Returns:
+    ///  The name of the layer
+    #[getter]
+    fn layer_name(&self) -> PyGenericIterable {
+        let edges = self.builder.clone();
+        (move || edges().layer_name()).into()
+    }
```

Comment:
this should return StringIterable

Reply:
I am returning a custom type OptionArcStringIterable now